src/checkers.cpp

```diff
@@ -88,37 +88,45 @@ ada_really_inline constexpr uint8_t path_signature(
   size_t i = 0;
   uint8_t accumulator{};
   for (; i + 7 < input.size(); i += 8) {
-    accumulator |= uint8_t(path_signature_table[uint8_t(input[i])] |
-                           path_signature_table[uint8_t(input[i + 1])] |
-                           path_signature_table[uint8_t(input[i + 2])] |
-                           path_signature_table[uint8_t(input[i + 3])] |
-                           path_signature_table[uint8_t(input[i + 4])] |
-                           path_signature_table[uint8_t(input[i + 5])] |
-                           path_signature_table[uint8_t(input[i + 6])] |
-                           path_signature_table[uint8_t(input[i + 7])]);
+    accumulator |= static_cast<uint8_t>(
```

Comment:
Here you have significantly harmed the readability.... This is not good:

```
 accumulator |= static_cast<uint8_t>(
         path_signature_table[static_cast<uint8_t>(input[i])] |
         path_signature_table[static_cast<uint8_t>(input[i + 1])] |
         path_signature_table[static_cast<uint8_t>(input[i + 2])] |
         path_signature_table[static_cast<uint8_t>(input[i + 3])] |
         path_signature_table[static_cast<uint8_t>(input[i + 4])] |
         path_signature_table[static_cast<uint8_t>(input[i + 5])] |
         path_signature_table[static_cast<uint8_t>(input[i + 6])] |
         path_signature_table[static_cast<uint8_t>(input[i + 7])]);
```

Reply:
But the intent is more clear. Better way might be to have a separate function for this part:

```c++
path_signature_table[static_cast<uint8_t>(input[i + 1])]
```

But I don't agree with the use of c-style cast at all.